### DISCUSSION TITLE
unstable_getServerSession breaks jest tests

### DISCUSSION THREAD

**Discussion by elliotgonzalez-lk** [nextauthjs/next-auth#6084] [Help]
I have an application with components executing `unstable_getServerSession` inside of `getServerSideProps` in order to protect the specific route.  I prefer to keep this session check on the server, as opposed to using `useSession` on the client.  However, using `unstable_getServerSession` breaks jest when importing the component into a test file.  This behavior has been discussed in depth here: https://github.com/nextauthjs/next-auth/issues/4866.

I have followed the recommendation of running two different jest environments for my tests, and testing `getServerSideProps` in a node environment, mocking `unstable_getServerSession`, and my UI is tested in JSDOM.  This works.  However, in order to test the actual component UI, I have to extract ALL the JSX into another file/component, and import that into my `pages/index.tsx`.  This has become very tedious and is IMO a very awkward developer experience.  Has anyone come up with a better solution for this?  Any help would greatly be appreciated!

**Comment by DoctorDerek** [comment]
I agree that the best approach would be for the "next-auth" maintainers to support this simple workflow:
1. Developer writes a Jest test for a Next.js page
2. Test works even when using `unstable_getServerSession`

Currently the workaround is to use extensive mocking patched together from various sources -- see #4866.
